### PR TITLE
Bump lcp to v0.2.10

### PR DESCRIPTION
**Pull request by junichi-tanaka**
## Why
- ref: https://github.com/datachainlab/lcp/pull/108

## What Changed
- Bump the version of lcp to v0.2.10
- Apply changes similar to below
  - https://github.com/datachainlab/ethereum-elc/pull/19
  - https://github.com/datachainlab/cosmos-ethereum-ibc-lcp/pull/47
- Fix lint errors

### CODE REVIEW

File: e2e/Enclave.mk [hunk 1]
```diff
@@ -130,6 +129,6 @@ $(Signed_RustEnclave_Name): $(RustEnclave_Name)
 
 .PHONY: enclave
 enclave:
-	@cd enclave && RUSTFLAGS=$(RUSTFLAGS) cargo build $(CARGO_TARGET) $(CARGO_FEATURES)
+	cd enclave && RUSTFLAGS=$(RUSTFLAGS) cargo build $(CARGO_TARGET) $(CARGO_FEATURES)
```
Comment (bluele) [inline e2e/Enclave.mk:132]: Is this change needed?

Comment (junichi-tanaka) [reply e2e/Enclave.mk:132]: Sorry, it's my mistake. I'll revert it. 🙇🏼‍♂️

File: e2e/Enclave.mk [hunk 2]
```diff
@@ -52,7 +49,8 @@ endif
 
 SGX_COMMON_CFLAGS += -fstack-protector
 
-CARGO_FEATURES = --features=default
+ENCLAVE_CARGO_FEATURES = --features=default
+APP_CARGO_FEATURES     = --features=default
```
Comment (bluele) [inline e2e/Enclave.mk:53]: Looks like `APP_CARGO_FEATURES` is unused and not necessary. I think these changes (*_CARGO_FEATURES) are unnecessary.

File: e2e/.ignore [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# cargo-machete v0.7.0 detects unused dependencies in lcp.
+# Thus ignore linting for submodules until resolved the issue.
+# ---
+# Analyzing dependencies of crates in this directory...
+# cargo-machete found the following unused dependencies in this directory:
+# light-client -- ./e2e/lcp/modules/light-client/Cargo.toml:
+#         serde
+lcp/
```
Comment (bluele) [inline e2e/.ignore:8]: I didn't realize this. In cases where the `ibc` feature is not enabled, `serde` may not be necessary.

File: e2e/Enclave.mk [hunk 4]
```diff
@@ -52,27 +49,20 @@ endif
 
 SGX_COMMON_CFLAGS += -fstack-protector
 
-CARGO_FEATURES = --features=default
```
Comment (bluele) [inline e2e/Enclave.mk:55]: @junichi-tanaka  Can you also remove `CARGO_FEATURES` reference in `enclave` target?

Comment (junichi-tanaka) [reply e2e/Enclave.mk:55]: @bluele removed in https://github.com/datachainlab/besu-qbft-elc/pull/8/commits/70106c3b5f83db74958ed86e6ce19e66771d5188

**Review by bluele — APPROVED** [review]
LGTM👍
